source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md

```diff
@@ -0,0 +1,185 @@
+```bash
+# ssh root@engine
+# cd /etc/httpd/saml2
+# curl -s -o idp-metadata.xml http://<keycloak-server>:8080/auth/realms/master/protocol/saml/descriptor
+# systemctl restart httpd
+```
+
+## oVirt AAA configuration
+In this part we create last peace which we need to get this up and running.
+
+### Installation
+```
+# yum install -y ovirt-engine-extension-aaa-ldap ovirt-engine-extension-aaa-misc
+```
+
+### Create /etc/ovirt-engine/extensions.d/http-authn.properties
+```
+ovirt.engine.extension.name = http-authn
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.misc
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.misc.http.AuthnExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Authn
+ovirt.engine.aaa.authn.profile.name = example-http
+ovirt.engine.aaa.authn.authz.plugin = ldap-authz
+ovirt.engine.aaa.authn.mapping.plugin = example-http-mapping
+config.artifact.name = HEADER
+config.artifact.arg = X-Remote-User
+```
+
+### Create /etc/ovirt-engine/extensions.d/example-http-mapping.properties
+```
+ovirt.engine.extension.name = example-http-mapping
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.misc
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.misc.mapping.MappingExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Mapping
+config.mapAuthRecord.type = regex
+config.mapAuthRecord.regex.mustMatch = false
+config.mapAuthRecord.regex.pattern = ^(?<user>.*?)((\\\\(?<at>@)(?<suffix>.*?)@.*)|(?<realm>@.*))$
+config.mapAuthRecord.regex.replacement = ${user}${at}${suffix}
+```
+
+### Create /etc/ovirt-engine/extensions.d/ldap-authz.properties
+```
+ovirt.engine.extension.name = ldap-authz
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.ldap
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.ldap.AuthzExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Authz
+config.profile.file.1 = ../aaa/ldap.properties
+```
+
+### Create /etc/ovirt-engine/aaa/ldap.properties
+```
+include = <389ds.properties>
+
+vars.server = 389ds.example.com
+vars.user = uid=user1,ou=users,dc=389ds,dc=example,dc=com
+vars.password = password
+
+pool.default.auth.simple.bindDN = ${global:vars.user}
+pool.default.auth.simple.password = ${global:vars.password}
+pool.default.serverset.type = single
+pool.default.serverset.single.server = ${global:vars.server}
+```
+
+This is it, now restart ovirt-engine:
+```
+systemctl restart ovirt-engine.service
+```
+
+If you now try to login to webadmin, you will be redirected `Keycloak`, where if you didn't login before you can enter credentials from your 389ds ldap.
```

Comment:
If you try to login to webadmin, you will be redirected to `Keycloak` (if you are not already logged in), where you can enter credentials from your 389ds ldap setup.

Reply:
Done